include/server/mir/scene/surface.h

```diff
@@ -128,6 +128,12 @@ class Surface :
 
     virtual auto focus_state() const -> MirWindowFocusState = 0;
     virtual void set_focus_state(MirWindowFocusState focus_state) = 0;
+
+    /// Often the same as the session name, but on Wayland can be set on a per-window basis
+    /// See xdg_toplevel.set_app_id and http://standards.freedesktop.org/desktop-entry-spec/ for more details
+    /// Empty string if not set
+    virtual auto application_id() const -> std::string = 0;
+    virtual void set_application_id(std::string const& application_id) = 0;
```

Comment:
Is this comment intended to apply to both methods? That's not how doxygen behaves.

As far as "Empty string if not set" is concerned, it suggests a non-existent "not set" state. If we need to say anything, then "defaults to empty string" is probably clearer. (Or should there actually be a "not set" state?)

Reply:
Fixed (see full diff if it doesn't show up in the mini view for some reason)